test/testcafe/messagelist_tests.ts

```diff
@@ -187,7 +189,9 @@ function endCode(msg) {
 function MSGinViewportSelector(timeout = 2000): Selector {
   return Selector(
     () => {
-      let raw_labels = document.querySelectorAll('#message-list .msg-body')
+      let raw_labels = document.querySelectorAll(
+        '#message-list .msg-body, .join-button'
+      )
```

Comment:
we probably need something more advanced here as this can currently only detect video messages and does not necessarily select the last/most recent video message.

also does it work as expected? means that it only passes if the element is visible to the user without scrolling.

Reply:
https://testcafe.io/documentation/402829/guides/basic-guides/select-page-elements could help here, its the documentation about testcafe selectors

Reply:
As it is only for the test, we can decide that we won't have multiple videochat invitations in the chat. So in my opinion, it doesn't really matter if the selector theoretically selects all video chat invitations. 
For me it worked as expected. When the videochat invitation was not in sight for the user, the test failed, when I stopped and scrolled down, it passed.